### PR TITLE
Refactor area and turf lighting

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request
Pt1. of a bacon/fancy/hybrid/whatever lighting port

Replaces the current area and turf ligting system with one that uses an MA overlay instead whos color and intensity can be varedited and changed
Minorly changes the turf lighting as well
Wouldnt mind a small tm to test as i might have missed an area or two

## Why It's Good For The Game

Allows for more flexible area lighting with cleaner code

## Changelog
:cl:
refactor: Area based lighting can be varedited and changed now
/:cl:

### CODE REVIEW

**Comment by TiviPlus** [comment]
@Kylerace btw

File: code/datums/mapgen/JungleGenerator.dm [hunk 1]
```diff
@@ -87,5 +87,5 @@
 
 /area/mine/planetgeneration
 	name = "planet generation area"
-	dynamic_lighting = DYNAMIC_LIGHTING_DISABLED
+	static_lighting = FALSE
```
Comment (Kylerace) [inline code/datums/mapgen/JungleGenerator.dm:90]: why did not having DYNAMIC lighting in the old system transition into not having STATIC lighting in the new system?

File: code/game/area/areas/away_content.dm [hunk 2]
```diff
@@ -14,21 +14,15 @@ Unused icons for new areas are "awaycontent1" ~ "awaycontent30"
...
 
 /area/awaymission/errorroom
 	name = "Super Secret Room"
-	dynamic_lighting = DYNAMIC_LIGHTING_DISABLED
+	static_lighting = FALSE
+	base_lighting_alpha = 255
+	base_lighting_color = COLOR_WHITE
 	has_gravity = STANDARD_GRAVITY
 
-/area/awaymission/vr
```
Comment (Kylerace) [inline code/game/area/areas/away_content.dm:27]: is this not used by an away mission?

**Review by Kylerace — CHANGES_REQUESTED** [review]

File: code/game/area/areas/away_content.dm (hunk 2 above)
Comment (TiviPlus) [reply code/game/area/areas/away_content.dm:27]: Nope

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -42,7 +42,7 @@ GLOBAL_LIST_EMPTY(station_turfs)
 	///Lumcount added by sources other than lighting datum objects, such as the overlay lighting component.
 	var/dynamic_lumcount = 0
 
-	var/dynamic_lighting = TRUE
+	var/always_lit = FALSE
```
[the remark this answers is not in the document]
Comment (TiviPlus) [reply code/game/turfs/turf.dm:45]: done

File: code/datums/mapgen/JungleGenerator.dm (hunk 1 above)
Comment (TiviPlus) [reply code/datums/mapgen/JungleGenerator.dm:90]: Before it was
DYNAMIC_LIGHTING_DISABLED = always lit
DYNAMIC_LIGHTING_ENABLED = genertes lighting objects unless the area is an area that does not use power
DYNAMIC_LIGHTING_FORCED = genertes lighting objects
DYNAMIC_LIGHTING_IFSTARLIGHT genertes lighting objects depending on if the config is enabled

This has been Separated into
static_lighting : whether lighting objects get generated
base_lighting_alpha+color: how lit the area always is (ie current behavior is white at 255 alpha)
starlight lighting flag which sets static_lighting in Initialize and is checked in other starlight checks

**Comment by AMonkeyThatCodes** [comment]
this it not a refactor but it's a cool pr so I don't care 😎

**Pull request merged**